File: frontend/src/types/event/event.ts
import { EventType } from './event-type';

export interface Event {
  id: string;
  name: string;
  description?: string;
  startDate: Date | null;
  endDate?: Date | null;
  location?: string;
  familyId: string | null;
  relatedMembers?: string[];
  type: EventType;
  color?: string; // Added color property
}

export interface EventFilter {
  searchQuery?: string;
  type?: EventType;
  eventType?: EventType;
  familyId?: string | null ;
  startDate?: Date | null;
  endDate?: Date | null;
  location?: string;
  relatedMemberId?: string;
}

File: backend/src/Domain/Entities/Event.cs
using backend.Domain.Enums;

namespace backend.Domain.Entities;

public class Event : BaseAuditableEntity
{
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public string? Location { get; set; }
    public Guid? FamilyId { get; set; }
    public EventType Type { get; set; }
    public string? Color { get; set; }

    public ICollection<Member> RelatedMembers { get; set; } = new List<Member>();
}
